tools/fetch-distro.py

```diff
@@ -60,35 +60,55 @@ def checkout_distro(args, distro: str, config: dict):
 
     url = config['Environment']['GIT_URL']
     branch = config['Environment']['GIT_BRANCH']
+    subdir = config['Environment']['GIT_SUBDIR'] if 'GIT_SUBDIR' in config['Environment'] else None
+
+    # Do not checkout the full sources if the package is in a subdirectory,
+    # a sparse checkout will be done after
+    sparse = ['--no-checkout', '--filter=blob:none'] if subdir is not None else []
 
     # Only debian uses source-git for now…
-    reference = [f'--reference-if-able=.'] if distro == 'debian' else []
+    reference = ['--reference-if-able=.'] if distro == 'debian' else []
 
     cmd = [
         'git', 'clone', url,
         f'--branch={branch}',
+        *sparse,
         dest.as_posix(),
         *reference,
     ]
     print(f"+ {shlex.join(cmd)}")
     subprocess.check_call(cmd)
 
+    # Sparse checkout if the package is in a subdirectory
+    if subdir is not None:
+        cmd = ['git', '-C', f'pkg/{distro}', 'sparse-checkout', 'set',
+               '--no-cone', f'{subdir}']
+        print(f"+ {shlex.join(cmd)}")
+        subprocess.check_call(cmd)
+
+        cmd = ['git', '-C', f'pkg/{distro}', 'checkout', 'HEAD']
+        print(f"+ {shlex.join(cmd)}")
+        subprocess.check_call(cmd)
+
     args.fetch = False  # no need to fetch if we just cloned
 
 def update_distro(args, distro: str, config: dict):
     branch = config['Environment']['GIT_BRANCH']
+    subdir = [config['Environment']['GIT_SUBDIR']] if 'GIT_SUBDIR' in config['Environment'] else []
```

Comment:
Ditto

Reply:
ok, this 2nd one required more adjustment